haystack/tools/tool.py

```diff
@@ -105,6 +116,22 @@ def __post_init__(self):
                 raise ValueError("outputs_to_string source must be a string.")
             if "handler" in self.outputs_to_string and not callable(self.outputs_to_string["handler"]):
                 raise ValueError("outputs_to_string handler must be callable")
+            if "source" in self.outputs_to_string or "handler" in self.outputs_to_string:
+                for key in self.outputs_to_string:
+                    if key not in {"source", "handler"}:
+                        raise ValueError(
+                            "Invalid outputs_to_string config. "
+                            "When using 'source' or 'handler' at the root level, no other keys are allowed. "
+                            "Use individual output configs instead."
+                        )
+            else:
+                for key, config in self.outputs_to_string.items():
```

Comment:
Could we add a dev comment here?
```suggestion
            else:
                # Multiple outputs configuration
                for key, config in self.outputs_to_string.items():
```